### PR TITLE
[FLINK-11126][YARN][security] Filter out AMRMToken in TaskManager‘s credentials

### PR DESCRIPTION
**Pull request by link3280**
## What is the purpose of the change

Currently, Flink JobManager propagates its storage tokens to TaskManager to meet the requirement of YARN log aggregation (see FLINK-6376). But in this way the AMRMToken is also included in the TaskManager credentials, which could be potentially insecure. 

The PR filters out AMRMToken before setting the tokens to TaskManager's container launch context, and adds checks for delegation tokens that JobManager prepares for TaskManagers.

## Brief change log

  - *Filter out AMRMToken before setting the tokens to the TaskManager container context.*

## Verifying this change

This change added tests and can be verified as follows:

  - *Extended YarnApplicationMasterRunnerTest to check delegation tokens in the TaskManager executor context.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: yes
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented?

### CODE REVIEW

File: flink-yarn/src/test/java/org/apache/flink/yarn/YarnApplicationMasterRunnerTest.java [hunk 1]
```diff
@@ -108,5 +131,39 @@ public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
...
+				hasHdfsDelegationToken = true;
+			}
+		}
+		assertTrue(hasHdfsDelegationToken);
+		assertFalse(hasAmRmToken);
+	}
+
+	// A helper method used for setting system env variables for the test.
+	// See https://stackoverflow.com/a/496849/6558955
+	private static void setEnv(Map<String, String> newEnv) throws Exception {
```
Comment (klion26) [inline flink-yarn/src/test/java/org/apache/flink/yarn/YarnApplicationMasterRunnerTest.java]: Maybe you could use `org.apache.flink.core.testutils.CommonTestUtils#setEnv`

Comment (link3280) [reply flink-yarn/src/test/java/org/apache/flink/yarn/YarnApplicationMasterRunnerTest.java]: @klion26 Thanks for the pointer! I'll try it.

**Comment by link3280** [comment]
The build failed with some network issues. I'll close and reopen the PR to trigger a rebuild.

**Comment by link3280** [comment]
@zentol @tzulitai Please have a look. ✋

**Comment by link3280** [comment]
@tillrohrmann @aljoscha PTAL. Should be a quick one.

**Pull request closed without merging**